### ISSUE TITLE
Expo start --web broken `symbolicateMiddleware`

### ISSUE THREAD

**Issue by Slamerz** [infinitered/ignite#2313]
### Describe the bug

Created 2 fresh projects of ignite, both show the same behavior when you attempt to run for web.

Seems to be an issue with [Expo-CLI](https://githubplus.com/expo/expo-cli/issues/4602)

Current Error.
 ```
$ expo start --web
Starting project at C:\Users\x\WebstormProjects\ignite8
Starting Metro Bundler
Error: Cannot find module '@expo/dev-server/build/webpack/symbolicateMiddleware'
Require stack:
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\webpack\WebpackBundlerDevServer.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\DevServerManager.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\startAsync.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\index.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\bin\cli
Error: Cannot find module '@expo/dev-server/build/webpack/symbolicateMiddleware'
Require stack:
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\webpack\WebpackBundlerDevServer.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\DevServerManager.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\startAsync.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\index.js
- C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\bin\cli
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\webpack\WebpackBundlerDevServer.js:6:30)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.webpack (C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\DevServerManager.js:41:18)
    at DevServerManager.startAsync (C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\server\DevServerManager.js:133:63)
    at async startAsync (C:\Users\x\WebstormProjects\ignite8\node_modules\@expo\cli\build\src\start\startAsync.js:115:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

Steps to reproduce.
Create a fresh ignite project
`npx ignite-cli@latest new PizzaApp --yes`
Navigate to project
`npm run expo:web`

### Ignite version

8.4.0

### Additional info

```
System
  platform           win32                                                        
  arch               x64         
  cpu                12 cores     AMD Ryzen 5 3600 6-Core Processor              
  directory          ignite8      C:\Users\x\WebstormProjects\ignite8

JavaScript (and globally-installed packages)
  node                 16.16.0      C:\Program Files\nodejs\node.EXE
  npm                  8.16.0       C:\Program Files\nodejs\npm.CMD
    corepack           0.10.0      
    expo-cli           6.0.2       
    ignite-cli         7.14.1      
    npm                8.16.0      
    yarn               1.22.19     
  yarn                 1.22.19      C:\Program Files\nodejs\yarn.CMD
    create-react-app   3.4.1       
    ignite-cli         3.5.3       
    react-native-cli   2.0.1       
  pnpm                 -            not installed
  expo                 46.0.17      managed

Ignite
  ignite-cli         7.14.1       C:\Program Files\nodejs\ignite.CMD
  ignite src         build        C:\Users\x\AppData\Roaming\nvm\v16.16.0\node_modules\ignite-cli\build

Android
  java               19.0.1       C:\Program Files\Common Files\Oracle\Java\javapath\java.EXE
  android home       -            undefined
  ```

**Comment by frankcalise** [comment]
@Slamerz thanks for the report! we're almost done publishing an update for this, it's an issue upstream with Expo.

More details in this PR - https://github.com/infinitered/ignite/pull/2303, specifically this: https://github.com/expo/expo/issues/19833

**Comment by ryskin** [comment]
Yeah I have a same issue

**Comment by gutenfries** [comment]
I also have this issue...

**Comment by pecabum** [comment]
+1 for the same issue

**Comment by Gedrovits** [comment]
It is still happening on a fresh install, according to the guide.

**Ignite version**
8.4.2

**Additional info**
```
PS C:\Users\vjats\Documents\GitHub\Experiments\NamelessTacticalRpg> npx ignite-cli doctor
System
  platform           win32
  arch               x64
  cpu                16 cores              AMD Ryzen 7 3700X 8-Core Processor
  directory          NamelessTacticalRpg   C:\Users\vjats\Documents\GitHub\Experiments\NamelessTacticalRpg

JavaScript (and globally-installed packages)
  node               18.12.1      C:\Program Files\nodejs\node.EXE
  npm                8.19.2       C:\Program Files\nodejs\npm.CMD
    corepack         0.14.2
    npm              8.19.2
  yarn               -            not installed
  pnpm               -            not installed
  expo               46.0.17      managed

Ignite
  ignite-cli         8.4.2        C:\Users\vjats\AppData\Local\npm-cache\_npx\e31027f3785124a8\node_modules\.bin\ignite.CMD
  ignite src         build        C:\Users\vjats\AppData\Local\npm-cache\_npx\e31027f3785124a8\node_modules\ignite-cli\build

Android
  java               -            null
  android home       -            undefined

Tools
  git                git version 2.38.1.windows.1   C:\Program Files\Git\cmd\git.EXE
```

**Comment by aaronsmulktis** [comment]
@Gedrovits hey that worked, thank you!

Tho still having trouble rendering something in the root tag, and I've tried several things within `index.js`:

```
import { registerRootComponent } from 'expo';
import { AppRegistry, Platform } from 'react-native';
import App from './App';

AppRegistry.registerComponent('main', () => App);

if (Platform.OS === 'web') {
    const rootTag = document.getElementById('root') || document.getElementById('X');
    AppRegistry.runApplication('App', { rootTag });
    // registerRootComponent(App);
}
```

Error I'm seeing in browser console:
`Uncaught ReferenceError: process is not defined`

**Comment by mradenovic** [comment]
This still doesn't work on a fresh install with `npm`.  The issue is caused by `package.json` `resolutions` field. Unlike `yarn`, `npm` will not install those dependencies.

```json
{
  ...
  "resolutions": {
    "@types/react": "^17",
    "@types/react-dom": "^17",
    "@expo/dev-server": "0.1.120"
  },
  ...
}
```

If `yarn` is used as a package manager, runs out of the box.

**Comment by frankcalise** [comment]
@mradenovic if you `npm i @expo/dev-server@0.1.120` as mentioned above, does it work for you?

**Comment by mradenovic** [comment]
@frankcalise workaround does work and it will add the package to `dependencies`. The question is: How this should be addressed?

Move all packages from `resolutions` to `dependencies` (what are the consequences?) or force `yarn install` or prompt the user to do the additional install with `npm`.

**Comment by frankcalise** [comment]
@mradenovic thanks for the feedback, the intent is that it works with npm if that is your main package manager. We'll look into it. If you have any other useful info along the way please share it.

Thanks for testing.

**Comment by mradenovic** [comment]
After some research my understanding is that  `npm` `overrides` and `yarn` `resolutions` provide the same/similar functionality: overriding dependencies even if they are nested. In this case, it was probably intended to use a known working `@expo/dev-server`.

The package should be added in all three sections: `devDependencies` (to make sure it will be installed), `resolutions` and `overrides` (to make sure the right version will be installed).

**Comment by mradenovic** [comment]
Just tested this. Everything that's in `resolutions` should probably be in `overrides` too, and vice versa. There is one package in `overrides` that probably will not be the right version if `yarn install` is used.

```json
  "overrides": {
    "react-error-overlay": "6.0.9"
  },

```

**Comment by Unforgiven-wanda** [comment]
As of today, this is still broken on a fresh install.
Any fix?

**Comment by frankcalise** [comment]
@mradenovic adding this to devDependencies and overrides did not seem to make a difference for me when using NPM, I still received the error.

**Comment by frankcalise** [comment]
I believe this should be closed by #2377, make sure to update and give it a try!

Feel free to reopen if still an issue
